helm-charts/support/values.yaml

```diff
@@ -1,30 +1,83 @@
+# added or removed. A pending pod is a sign that another node should be added,
+# and an underused node is a sign that a node should be removed.
+#
+# We always need a cluster-autoscaler, but only deploy it ourselves if its not
+# provided as part of the k8s cluster.
+#
+# values ref: https://github.com/kubernetes/autoscaler/blob/master/charts/cluster-autoscaler/values.yaml
+#
 cluster-autoscaler:
   enabled: false
+
+# ingress-nginx is responsible for proxying traffic based on k8s Ingress
+# resources defined in the k8s cluster.
+#
+# Typically, all inbound traffic arrives to some ingress-nginx controller pod
+# via a k8s Service that has a public IP, and is thereafter proxied to a k8s
+# Service (a Pod selected by a k8s Service) that doesn't have a public IP.
+#
+# values ref: https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/values.yaml
+#
 ingress-nginx:
   controller:
     podLabels:
-      # nginx-ingress controllers need to communicate with the proxy pod (and only the proxy pod)
-      # in clusters with networkpolicy enforcement enabled. Adding this label on the controller pod
-      # allows that. https://z2jh.jupyter.org/en/stable/administrator/security.html#introduction-to-the-chart-s-four-network-policies
-      # has more information.
+      # nginx-ingress controllers need to be allowed proxy traffic onwards to
+      # JupyterHub's proxy pod (and only the proxy pod) in clusters with
+      # NetworkPolicy enforcement enabled. Adding this label on the controller
+      # pod allows that.
+      #
+      # ref: https://z2jh.jupyter.org/en/stable/administrator/security.html#introduction-to-the-chart-s-four-network-policies
+      #
       hub.jupyter.org/network-access-proxy-http: "true"
+
+# prometheus is responsible for collecting metrics and informing grafana about
+# the metrics on request. It comes with several dependency charts, where we
+# opt-out of a few.
+#
+# values ref: https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/values.yaml
+#
 prometheus:
-  networkPolicy:
-    enabled: true
+  # alertmanager is an optional prometheus chart dependency that we opt-out from
+  # as we favor Grafana for this functionality. Grafana provides alerts and does
+  # so with a better UI that we expose publicly behind auth anyhow.
+  #
   alertmanager:
-    # We do not use the Prometheus alertmanager component, as we already have Grafana
-    # Grafana also has alerts, and the UI of grafana is much better than that of alertmanager
-    # We also expose Grafana publicly behind auth anyway, so we can consolidate alerting with Grafana too
     enabled: false
-  nodeExporter:
+
+  # prometheus-pushgateway is an optional prometheus chart dependency that we
+  # opt-out from. pushgateway provides a way to complement prometheus server's
+  # behavior of scraping metrics from services by allowing services to push
+  # metrics to prometheus.
+  #
+  prometheus-pushgateway:
+    enabled: false
```

Comment:
The actual migration in config includes renaming this from `pushgateway` to `prometheus-pushgateway`.

I also removed a redundant declaration of using RollingUpgrade for the node-exporter chart.